### PR TITLE
BulletPhysics

### PR DESCRIPTION
**Pull request by VZout**

### CODE REVIEW

File: src/model_pool.hpp [hunk 1]
```diff
@@ -254,7 +256,7 @@ namespace wr
 			return nullptr;
 		}
 
-		loader->DeleteModel(data);
+		//loader->DeleteModel(data);
```
Comment (LaisoEmilio) [inline src/model_pool.hpp]: Commented code? Can it be removed or is it just a temporary change?

File: src/model_pool.hpp [hunk 2]
```diff
@@ -396,7 +399,7 @@ namespace wr
 			return nullptr;
 		}
 
-		loader->DeleteModel(data);
+	//	loader->DeleteModel(data);
```
Comment (LaisoEmilio) [inline src/model_pool.hpp]: same as above

File: tests/demo/physics_engine.cpp [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+				if (!node->m_rigid_bodys.has_value() && node->m_rigid_body)
+				{
+					auto world_position = node->m_rigid_body->getWorldTransform().getOrigin();
+					node->m_position = util::DXV3toBV3(world_position);
+					node->SignalTransformChange();
+				}
+			}
+		}
+
+		/*auto& player_cam = std::dynamic_pointer_cast<DebugCamera>(sg.GetActiveCamera());
```
Comment (tntmeijs) [inline tests/demo/physics_engine.cpp]: May want to move some of this commented-out code to keep things nice and tidy.

File: tests/demo/physics_engine.hpp [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+namespace wr
+{
+	struct Model;
+}
+
+namespace phys
+{
+	namespace util
+	{
+		inline btVector3 BV3toDXV3(DirectX::XMVECTOR v)
```
Comment (tntmeijs) [inline tests/demo/physics_engine.hpp]: Wouldn't this be DXV3toBV3 instead?

File: tests/demo/scene_spheres.hpp [hunk 5]
```diff
@@ -76,7 +76,7 @@ namespace spheres_scene
 		return (1.0f - t) * v0 + t * v1;
 	}
 
-	void CreateScene(wr::SceneGraph* scene_graph, wr::Window* window)
+	void CreateScene(wr::SceneGraph* scene_graph, wr::Window* window, phys::PhysicsEngine& phys_engine)
 	{
 		//wr::Model* spheres[49];
```
Comment (tntmeijs) [inline tests/demo/scene_spheres.hpp]: Don't think this needs to stay commented-out.

File: tests/demo/scene_viknell.hpp [hunk 6]
```diff
@@ -138,11 +152,13 @@ namespace viknell_scene
 	{
 		t += 10.f * ImGui::GetIO().DeltaTime;
 
+		test_model->m_rigid_body->activate(true);
+
 		//auto pos = test_model->m_position;
```
Comment (tntmeijs) [inline tests/demo/scene_viknell.hpp]: Remove unused code.

File: tests/demo/scene_spheres.hpp (hunk 5 above)
Comment (VZout) [reply tests/demo/scene_spheres.hpp]: didn't change that code.

File: tests/demo/scene_viknell.hpp (hunk 6 above)
Comment (VZout) [reply tests/demo/scene_viknell.hpp]: didn't change that code.

File: tests/demo/physics_engine.hpp (hunk 4 above)
Comment (VZout) [reply tests/demo/physics_engine.hpp]: jep changed.

File: tests/demo/physics_engine.cpp (hunk 3 above)
Comment (VZout) [reply tests/demo/physics_engine.cpp]: done

**Pull request merged**